core/src/test/scala/pickling/run/runtime-instantiation-static-picklers.scala

```diff
@@ -0,0 +1,219 @@
+            b.hintElidedType(implicitly[FastTypeTag[T]])
+            // Ask directly to the scope instead of relying on `p`
+            val pickler = implicitly[Pickler[T]]
+            pickler.pickle(picklee.value, b)
+          })
+          builder.endEntry()
+
+        }
+
+        def unpickle(tag: String, reader: PReader) = {
+
+          val reader1 = reader.readField("value")
+          reader1.hintElidedType(implicitly[FastTypeTag[T]])
+          val tag1 = reader1.beginEntry()
+            // Ask directly to the scope instead of relying on `u`
+          val unpickler = implicitly[Unpickler[T]]
+          val result = unpickler.unpickle(tag1, reader1).asInstanceOf[T]
+          reader1.endEntry()
+
+        }
+      }
+
+}
+
+class StaticJavaRuntimeCustomPickler extends FunSuite {
+
+  implicit val staticOnly = static.StaticOnly
+  import ContainerPickling.containerPicklerUnpickler
+
+  test("runtime instantiated custom pickler should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val containerPickler = containerPicklerUnpickler[List[Int]]
+    val pickled = c.pickle.value
+
+    val picklerClassName = containerPickler.getClass.getName
+    val runtimePickler = RuntimeHelper.getInstance[Pickler[Container[Any]]](picklerClassName)
+    val format = implicitly[PickleFormat]
+    val unknownContainer = c.asInstanceOf[Container[Any]]
+    val pickled2 = runtimePickler.pickle(unknownContainer, format.createBuilder())
+    println(pickled)
+    println(pickled2)
+
+    assert(pickled === pickled2)
+
+  }
+
+  test("runtime instantiated custom unpickler should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val pickled = c.pickle.value
+    val blob = JSONPickle(pickled)
+    val reader = pickleFormat.createReader(blob)
+
+    val containerUnpickler = containerPicklerUnpickler[List[Int]]
+    val unpicklerClassName = containerUnpickler.getClass.getName
+    val runtimeUnpickler = RuntimeHelper.getInstance[Unpickler[Container[Any]]](unpicklerClassName)
+    
+    // Very important, assume it's elided
+    val tag = runtimeUnpickler.tag
+    assert(tag != null)
+
+    val unpickled = runtimeUnpickler.unpickle(tag.key, reader)
+    assert(c === unpickled)
+
+  }
+
+}
+
+class StaticJavaRuntimeCustomPicklerImplicitly extends FunSuite {
+
+  implicit val staticOnly = static.StaticOnly
+  import ContainerPickling.containerPicklerUnpicklerImplicitly
+
+  test("runtime instantiated custom pickler calling implicitly should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val containerPickler = containerPicklerUnpicklerImplicitly[List[Int]]
+    val pickled = c.pickle.value
+
+    val picklerClassName = containerPickler.getClass.getName
+    val runtimePickler = RuntimeHelper.getInstance[Pickler[Container[Any]]](picklerClassName)
+    val format = implicitly[PickleFormat]
+    val unknownContainer = c.asInstanceOf[Container[Any]]
+    val pickled2 = runtimePickler.pickle(unknownContainer, format.createBuilder())
```

Comment:
Ok, this makes more sense now.  You're creating a pickler using "unsafe new instance", not even calling a default constructor.  You'll actually have to *PICKLE* the pickler to make sure all of its state is safely written/restored, or create something which can look it up by string.

The NPE is actually perfectly expected since you're unsafe allocating  anew instance, and then calling methods on that instance.  I don't expect this to change, at a minimum you'd still have to call the default constructor after "new instancing".

In any case, I think you want to look at how Any{Un}Pickler actually works.   I still have some refactoring to make it cleaning, but follow the trail of logic here.   AnyPickler FIRST looks up any registered static picklers, then uses them.  You should be able to register your picklers with the runtime lookup and "AnyUnpickler" the values later.

Caveat:  I'm still working out all the kinks to make this 100% seamless.